app/controllers/admin/lead_provider_users_controller.rb

```diff
@@ -2,17 +2,22 @@
 
 class Admin::LeadProviderUsersController < Admin::BaseController
   before_action :set_lead_provider
+  skip_after_action :verify_authorized, only: :index
+  skip_after_action :verify_policy_scoped, except: :index
 
   def index
-    @users = @lead_provider.users
+    @users = policy_scope(@lead_provider.users)
   end
 
   def new
     @user = User.new
+    authorize @user
   end
 
   def create
-    @user = User.new(user_params)
+    @user = User.new(permitted_attributes(User))
+    authorize @user
+    authorize LeadProviderProfile
```

Comment:
What does `authorize` do exactly? 

Check what the `current_user` can do the things that are authorised?

Reply:
Something along the lines of `LeadProviderProfilePolicy.new(LeadProviderProfile, current_user).create?`. Probably worth reading over the [Pundit docs](https://github.com/varvet/pundit)